### PR TITLE
feat(slider): sass refactor, added story

### PR DESCRIPTION
**Pull request by sstrubberg**
Refs #8899 

Brings `slider` to the `carbon-react` package. Also brings `slider` into `@carbon/styles`.

#### Changelog

**New**

- Styles added for `Slider`
- Stories added for `Slider`

#### Testing / Reviewing

Pull down the PR locally, go to `carbon-react` folder and run `yarn storybook`. Ensure `Slider` renders properly.

Run `yarn test --watch packages/styles/scss/components/__tests__/slider-test.js` and make sure you get green checkmarks.

#### Current Bugs

Styles are pretty off. Any feedback is appreciated.

image

### CODE REVIEW

**Comment by jnm2377** [comment]
Do you wanna add the text input styles too? I did that with pagination, since it was using select styles. Just to make sure everything is working correctly... there were a few things I had to fix after adding in those missing styles, so I think it would be good to add it in this PR since they're related. 😬

**Comment by sstrubberg** [comment]
> Do you wanna add the text input styles too? I did that with pagination, since it was using select styles. Just to make sure everything is working correctly... there were a few things I had to fix after adding in those missing styles, so I think it would be good to add it in this PR since they're related. 😬

I can't add them until I've merged #8988, right?

**Comment by tw15egan** [comment]
Marked it blocked until #8988 is merged, will review that one now 👍🏻

**Comment by tw15egan** [comment]
Should be good to update now that #8988 has been merged 👍🏻

File: packages/carbon-react/src/components/Slider/Slider.stories.js
Comment (tw15egan) [inline packages/carbon-react/src/components/Slider/Slider.stories.js:39]: Can we remove this since we're using the props directly?

Comment (tw15egan) [inline packages/carbon-react/src/components/Slider/Slider.stories.js]: Can we remove this since we're using the props directly?

File: packages/carbon-react/src/components/Slider/Slider.stories.js
Comment (tw15egan) [inline packages/carbon-react/src/components/Slider/Slider.stories.js]: We can remove these since we are using the props directly, and not using the actions / knobs

File: packages/carbon-react/src/components/Slider/Slider.stories.js
Comment (tw15egan) [inline packages/carbon-react/src/components/Slider/Slider.stories.js]: We can remove the playground story since the knobs don't work

File: packages/styles/scss/components/slider/_slider.scss
Comment (tw15egan) [inline packages/styles/scss/components/slider/_slider.scss]:
```suggestion
@use '../form';
@use '../text-input';
```

File: packages/styles/scss/components/slider/_slider.scss
Comment (tw15egan) [inline packages/styles/scss/components/slider/_slider.scss:22]: I think the main reason the styles are off is that we are using the `rem` function, but it's not being imported. Sass functions that aren't found just fail silently, instead of giving us an error (which is annoying!) 

`@use '../../utilities/convert' as *;`

File: packages/styles/scss/components/slider/_slider.scss
Comment (tw15egan) [inline packages/styles/scss/components/slider/_slider.scss]:
```suggestion
/// @access public
```

**Review by tw15egan — CHANGES_REQUESTED** [review]
A few changes, but looking great!

**Comment by sstrubberg** [comment]
Got it all fixed up. Thx @tw15egan

**Review by tw15egan — APPROVED** [review]
Looks great!

**Review by jnm2377 — APPROVED** [review]

**Pull request merged**
